src/coreclr/jit/gentree.cpp

```diff
@@ -3545,6 +3545,34 @@ GenTree* Compiler::gtReverseCond(GenTree* tree)
     return tree;
 }
 
+/*****************************************************************************
+ *
+ *  Swap the meaning of the given comparison. Preserves GTF_RELOP_NAN_UN if set.
+ */
+
+GenTree* Compiler::gtSwapRelop(GenTree* tree, GenTree::ValueNumberUpdate vnUpdate)
+{
+    assert(tree->OperIsCompare());
+
+    bool isUnordered = false;
+
+    if (varTypeIsFloating(tree->gtGetOp1()))
+    {
+        isUnordered = (tree->gtFlags & GTF_RELOP_NAN_UN);
+    }
+
+    tree->SetOper(GenTree::SwapRelop(tree->OperGet()), vnUpdate);
+
+    // SetOper will clear the GTF_RELOP_NAN_UN bit.
+    // Therefore, we need to set it again if it was set before.
```

Comment:
That does not sound true, `SetOper` does not clear oper-specific flags (it is inteded to be used when switching between opers of the same "class").

Reply:
You are right. I was thinking of `ChangeOper`. Now that I'm thinking about this, I should revert this change. In the place where I used `ChangeOper`, I should use `SetOper`.

Having `SetOper` and `ChangeOper` is a little confusing to me.